src/components/gdg/End.tsx

```diff
@@ -64,6 +68,43 @@ function InfoSection({
 }
 
 export default function End(): JSX.Element {
+  const [events, setEvents] = useState<EventData[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const org = "gdg";
+  const orderByProximity = true;
+  // const proximity = 5;
+
+  // Construct the query parameters:
+  const params = new URLSearchParams();
+  params.append("orderByProximity", orderByProximity.toString());
+  // if (proximity) {
+  //   params.append("proximity", proximity.toString());
+  // }
+
+  useEffect(() => {
+    const fetchGDGEvents = async () => {
+      setIsLoading(true);
+      try {
+        const response = await fetch(`/api/events/${org}?${params.toString()}`);
+        console.log(response);
+        if (!response.ok) {
+          throw new Error(`ERROR: ${response.status}`);
+        }
+
+        const data: EventData[] = await response.json();
+        console.log(data);
+        setEvents(data);
+      } catch (error: any) {
+        console.error(error.message);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchGDGEvents();
+  }, []);
```

Comment:
looking good, if you are interested in a more production-friendly way of implementing this, checkout react query, but this implementation is fine for this project.